gomilter.go

```diff
@@ -17,127 +17,128 @@ package gomilter
 #cgo LDFLAGS: -lmilter
 
 #include <stdlib.h>
-//#include <string.h>
-#include "libmilter/mfapi.h" 
+#include <arpa/inet.h>
+#include "libmilter/mfapi.h"
 #include "filter.h"
 
 */
 import "C"
 import (
-  "fmt"
-  "unsafe"
-  "encoding/binary"
-  "reflect"
-  "os"
-  "strings"
-  "bytes"
-  "encoding/gob"
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"fmt"
+	"net"
+	"os"
+	"reflect"
+	"strings"
+	"unsafe"
 )
 
 type sockaddr_in struct {
-  sin_family int8
-  sin_port uint8
-  sin_addr uint32
-  sin_zero uint64
+	sin_family int8
+	sin_port   uint8
+	sin_addr   uint32
+	sin_zero   uint64
 }
 
 // Return values for Callback functions
 const (
-  Continue = iota
-  Reject
-  Discard
-  Accept
-  Tempfail
-  _
-  _
-  Noreply
-  Skip
+	Continue = iota
+	Reject
+	Discard
+	Accept
+	Tempfail
+	_
+	_
+	Noreply
+	Skip
 )
 
 // flags
 const (
-  ADDHDRS     = 0x00000001 // 000000001
-  CHGBODY     = 0x00000002 // 000000010
-  ADDRCPT     = 0x00000004 // 000000100
-  DELRCPT     = 0x00000008 // 000001000
-  CHGHDRS     = 0x00000010 // 000010000
-  QUARANTINE  = 0x00000020 // 000100000
-  CHGFROM     = 0x00000040 // 001000000
-  ADDRCPT_PAR = 0x00000080 // 010000000
-  SETSYMLIST  = 0x00000100 // 100000000
+	ADDHDRS     = 0x00000001 // 000000001
+	CHGBODY     = 0x00000002 // 000000010
+	ADDRCPT     = 0x00000004 // 000000100
+	DELRCPT     = 0x00000008 // 000001000
+	CHGHDRS     = 0x00000010 // 000010000
+	QUARANTINE  = 0x00000020 // 000100000
+	CHGFROM     = 0x00000040 // 001000000
+	ADDRCPT_PAR = 0x00000080 // 010000000
+	SETSYMLIST  = 0x00000100 // 100000000
 )
 
 // Interface that must be implemented in order to use gomilter
 type Milter interface {
-  GetFilterName() string
-  GetDebug() bool
-  GetFlags() int
-  GetSocket() string
+	GetFilterName() string
+	GetDebug() bool
+	GetFlags() int
+	GetSocket() string
 }
 
 // An "empty" Milter with no callback functions
 type MilterRaw struct {
-  FilterName string
-  Debug bool
-  Flags int
-  Socket string
+	FilterName string
+	Debug      bool
+	Flags      int
+	Socket     string
 }
 
 func (m *MilterRaw) GetFilterName() string {
-  return m.FilterName
+	return m.FilterName
 }
 
 func (m *MilterRaw) GetDebug() bool {
-  return m.Debug
+	return m.Debug
 }
 
 func (m *MilterRaw) GetFlags() int {
-  return m.Flags
+	return m.Flags
 }
 
 func (m *MilterRaw) GetSocket() string {
-  return m.Socket
+	return m.Socket
 }
 
 // ********* Callback checking types *********
 type checkForConnect interface {
-  Connect(ctx uintptr, hostname, ip string) (sfsistat int8)
+	Connect(ctx uintptr, hostname string, ip net.IP) (sfsistat int8)
```

Comment:
Probably good to mention that I did break BC here, by changing the type for ip from string to net.IP.